lib/chef/resource/sysctl_param.rb

```diff
@@ -0,0 +1,123 @@
+#
+# Copyright:: 2018, Webb Agile Solutions Ltd.
+# Copyright:: 2018, Chef Software Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "chef/resource"
+
+class Chef
+  class Resource
+    class SysctlParam < Chef::Resource
+      resource_name :sysctl_param
+      provides :sysctl_param
```

Comment:
I sort of think this should be:

```
resource_name :sysctl
provides :sysctl
provides :sysctl_param  # back compat with the cookbook
```

Reply:
Also even if we don't want to have two names, still don't need both `resource_name` and `provides` for the same sym.

Reply:
I pushed a rename commit